ckanext/asset_storage/plugin.py

```diff
@@ -22,8 +22,10 @@ def update_config(self, config):
         backend_config = config.get(uploader.CONF_BACKEND_CONFIG)
         if not backend_config:
             backend_config = {}
-        else:
+
+        if type(backend_config) == str:
             backend_config = literal_eval(backend_config)
+
```

Comment:
1. this should be an `elif` as there is no point in evaluating if we just set `backend_config` to a dict
2. It's preferrable `isinstance(x, foo)` and not `type(x) == foo` for type checking. This is because types can be subclassed, overridden etc. It's true that in this particular case it's not likely, but it's still a better choice (and IMHO more readable).
3. To make things absolutely and definitely compatible between Python 3.x and 2.7, using `six.string_types` is recommended

```suggestion
        import six  # put at the top of the file
        
        elif isinstance(backend_config, six.string_types):
            backend_config = literal_eval(backend_config)

```

Reply:
Thanks for the recommendations! I've implemented them :)